CMakeLists.txt

```diff
@@ -84,87 +80,68 @@ target_link_libraries(rmm INTERFACE spdlog::spdlog_header_only)
 target_link_libraries(rmm INTERFACE dl)
 target_compile_features(rmm INTERFACE cxx_std_17 $<BUILD_INTERFACE:cuda_std_17>)
 
-# Set logging level. Must go before including gtests and benchmarks.
-
-set(RMM_LOGGING_LEVEL
-    "INFO"
-    CACHE STRING "Choose the logging level.")
-# Set the possible values of build type for cmake-gui
-set_property(CACHE RMM_LOGGING_LEVEL PROPERTY STRINGS "TRACE" "DEBUG" "INFO" "WARN" "ERROR"
-                                              "CRITICAL" "OFF")
-message(STATUS "RMM: RMM_LOGGING_LEVEL = '${RMM_LOGGING_LEVEL}'")
-
 if((BUILD_TESTS OR BUILD_BENCHMARKS) AND CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
-  # Auto-detect available GPU compute architectures
-  include(${RMM_SOURCE_DIR}/cmake/Modules/SetGPUArchs.cmake)
-  # Enable the CUDA language after setting CMAKE_CUDA_ARCHITECTURES
+  include(rapids-cuda)
+  rapids_cuda_init_architectures(RMM)
   enable_language(CUDA)
+
+  # Since RMM only enables CUDA optionally we need to manually include the file that
+  # rapids_cuda_init_architectures relies on `project` calling
+  include("${CMAKE_PROJECT_RMM_INCLUDE}")
   message(STATUS "RMM: Building benchmarks with GPU Architectures: ${CMAKE_CUDA_ARCHITECTURES}")
 endif()
 
 # optionally build tests
-
 if(BUILD_TESTS AND CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
-  enable_testing()
+  include(cmake/thirdparty/get_gtest.cmake)
+  include(CTest) # calls enable_testing()
+
   add_subdirectory(tests)
 endif()
 
-# add google benchmark
-
+# optionally build benchmarks
 if(BUILD_BENCHMARKS AND CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+  include(cmake/thirdparty/get_gbench.cmake)
   add_subdirectory(benchmarks)
 endif()
 
-# install targets
+# install export targets
+install(TARGETS rmm EXPORT rmm-exports)
+install(DIRECTORY include/rmm/ DESTINATION include/rmm)
+install(FILES ${RMM_BINARY_DIR}/include/rmm/version_config.hpp DESTINATION include/rmm)
 
-include(GNUInstallDirs)
-set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/rmm)
+set(doc_string
+    [=[
+Provide targets for RMM: RAPIDS Memory Manager.
 
-install(TARGETS rmm EXPORT rmm-targets)
+The goal of the [RMM](https://github.com/rapidsai/rmm) is to provide:
 
-install(DIRECTORY include/rmm DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-install(FILES ${RMM_BINARY_DIR}/include/rmm/version_config.hpp
-        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/rmm)
+  A common interface that allows customizing device and host memory allocation
+  A collection of implementations of the interface
+  A collection of data structures that use the interface for memory allocation
+]=])
 
-include(CMakePackageConfigHelpers)
-configure_package_config_file(cmake/rmm-config.cmake.in ${RMM_BINARY_DIR}/cmake/rmm-config.cmake
-                              INSTALL_DESTINATION ${INSTALL_CONFIGDIR})
+set(code_string
+    [=[
+thrust_create_target(rmm::Thrust FROM_OPTIONS)
+]=])
```

Comment:
Can you explain what this is?

Reply:
The `[=[` , `]=]` are CMake's string literal syntax. It makes sure you don't need to escape any characters like `$` or `"`. I like to use the syntax where-ever possible to make sure we don't need to add escaping in the future.

The `thrust_create_target` command is how Thrust properly recreates the CMake target with both the correct backends ( github.com/NVIDIA/thrust/tree/main/thrust/cmake ) and to specify the correct include directories.  

Implicitly CMake treats all import targets include directories as system includes. This is done as it removes warnings from any of those includes. But for Thrust this is an issue as the CUDA Toolkit packages Thrust into a compiler implicit include directory, which takes higher precedence over user provided directories specified as system includes. The way to work around this in CMake requires jumping through some hoops ( to be fixed :: soon :: ) and that work is captured inside `thrust_create_target`

Reply:
If you meant how does `FINAL_CODE_BLOCK` for `rapids_export` work. The `rapids_export` command generates a `rmm-config.cmake` that consumers use via `find_package(rmm)`. Since this is based on a template file I added the `FINAL_CODE_BLOCK` as a way for projects to specify CMake code to be executed at the end of the config file.